### DISCUSSION TITLE
Why doesn't package.tools.make install a package with configs?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#2164] [Q&A]
https://github.com/xmake-io/xmake/blob/18b910e27d60ee244ee2245071f33ff0fc821991/xmake/modules/package/tools/make.lua#L142-L154

为啥这里运行`make`的时候传入configs参数，而`make install`不传参？直接使用Makefile没经过config，所有设置都必须从命令行传入，两次运行不传相同参数经常出问题（build static & install shared，报错找不到文件）

**Comment by waruqi** [comment]
build 用的参数 通常和 make install 参数不完全一直，大都情况下 `make XX= YY=; make install` 就够了，而且有些 build 传入 install ，反而会出错。。很多 CC= 什么的跟编译相关的参数 install 都用不到

默认如果全部同步传入，有时候反而出错，目前是由于现有 xmake-repo 仓库包 make install 不带参数大都足够满足需求了，所以我没加参数。。。

如果有特殊的参数需求，可以拆分单独走 make.build + make.make (install configs)

**Reply by xq114** [reply]
目前只有openssl和cmake两个包用了make.install，因为这两个包Makefile是由特殊的configure生成的，其他全都是手动拆分单独运行make的；感觉这里参数一致更加通用，参数不一致的再额外处理更好，并且额外处理起来也更简单

**Reply by waruqi** [reply]
因为现在参数必须一致  走 make install 的 case 太少，所以暂时没啥必要，等后面这种包多了，必须要进一步简化了，到时候根据现有包的情况，再考虑如何更好的简化封装。。现在暂时先这样吧
